pyuvdata/uvdata/uvdata.py

```diff
@@ -12246,6 +12609,15 @@ def write_miriad(
             raise ValueError("Cannot write out metadata only objects to a miriad file.")
 
         miriad_obj = self._convert_to_filetype("miriad")
+
+        # If we have a flex-pol dataset, remove it so that we can pass the data to the
+        # writer without issue. We create a copy because otherwise we run the risk of
+        # messing w/ the metadata of the original object, and because the overhead of
+        # doing so is generally smaller than that from removing flex-pol.
+        if self.flex_spw_polarization_array is not None:
+            miriad_obj = miriad_obj.copy()
```

Comment:
Does this copy run the risk of a large memory increase?

Reply:
It will increase memory, although I think the worst case scenario is that it will only do so by 50% (inflating a flex-pol dataset to "normal" size already balloons up the object quite a bit). I debated on whether to do something more sophisticated, but given that MIRIAD will never support this particular mode of writing out data, it seemed like I would be adding a fair bit of code to wrap what was an unusual corner case. The other "easy" alternative was to convert the original object back and forth between flex-pol, although I felt a little uneasy about changing the object like that on a write operation (especially since it means that if an error gets thrown, the object is left in it's supposedly transitory state). I could poke at this more if desired, although I think this path for the data is likely to be seldom used.

Reply:
I agree, I think it's unlikely to be heavily used. We can address it if and when it becomes a problem.